pkg/cloudprovider/plugins.go

```diff
@@ -40,10 +40,14 @@ var (
 		external bool
 		detail   string
 	}{
-		{"openstack", true, "https://github.com/kubernetes/cloud-provider-openstack"},
-		{"photon", false, "The Photon Controller project is no longer maintained."},
+		{"aws", false, "The AWS provider is deprecated and will be removed in the near future"},
```

Comment:
I think it would be worth either adding the KEP link or the sig cloud-provider link for background/details.

Reply:
I don't think that we should tie KEP locations to code.

Reply:
Thinking about how the deprecation works, if the goal is to deprecate every provider, we can enforce that in code more strongly. As it stands we opt-in to deprecation, provide a flag that indicates if an external provider is available, then send either the same message (aside from name) or point to a new repository.

Taking the DRY principle, for 1.13 I'm thinking the loader should automatically signal a deprecation with the named provider, and offer an external location if it is configured. In this way we guarantee a minimum deprecation warning, with the option to specify a migration path.